solana-openai/nodejs/src/index.ts

```diff
@@ -0,0 +1,172 @@
+    await litNodeClient.connect();
+    console.log("✅ Connected to the Lit network");
+
+    console.log("🔄 Connecting LitContracts client to network...");
+    const litContracts = new LitContracts({
+      signer: ethersWallet,
+      network: LIT_NETWORK,
+      debug: false,
+    });
+    await litContracts.connect();
+    console.log("✅ Connected LitContracts client to network");
+
+    if (LIT_PKP_PUBLIC_KEY === undefined || LIT_PKP_PUBLIC_KEY === "") {
+      console.log("🔄 PKP wasn't provided, minting a new one...");
+      pkpInfo = (await mintPkp(ethersWallet)) as {
+        tokenId?: string;
+        publicKey?: string;
+        ethAddress?: string;
+      };
+      console.log("✅ PKP successfully minted");
+      console.log(`ℹ️  PKP token ID: ${pkpInfo.tokenId}`);
+      console.log(`ℹ️  PKP public key: ${pkpInfo.publicKey}`);
+      console.log(`ℹ️  PKP ETH address: ${pkpInfo.ethAddress}`);
+    } else {
+      console.log(`ℹ️  Using provided PKP: ${LIT_PKP_PUBLIC_KEY}`);
+      pkpInfo = {
+        publicKey: LIT_PKP_PUBLIC_KEY,
+        ethAddress: ethers.utils.computeAddress(`0x${LIT_PKP_PUBLIC_KEY}`),
+      };
+    }
+
+    console.log("🔄 Creating AuthMethod using the ethersSigner...");
+    const authMethod = await EthWalletProvider.authenticate({
+      signer: ethersWallet,
+      litNodeClient,
+    });
+    console.log("✅ Finished creating the AuthMethod");
+
+    console.log("🔄 Getting the Session Signatures...");
+    const pkpSessionSigs = await litNodeClient.getPkpSessionSigs({
+      pkpPublicKey: pkpInfo.publicKey!,
+      chain: "ethereum",
+      authMethods: [authMethod],
+      expiration: new Date(Date.now() + 1000 * 60 * 10).toISOString(), // 10 minutes
+      resourceAbilityRequests: [
+        {
+          resource: new LitActionResource("*"),
+          ability: LitAbility.LitActionExecution,
+        },
+        {
+          resource: new LitPKPResource("*"),
+          ability: LitAbility.PKPSigning,
+        },
+      ],
+    });
+    console.log("✅ Generated the Session Signatures");
+
+    console.log("🔄 Generating wrapped key...");
+    const response = await generatePrivateKey({
+      pkpSessionSigs,
+      network: "solana",
+      memo: "This is a Dev Guide code example testing Solana key",
+      litNodeClient,
+    });
+    console.log(`✅ Generated wrapped key with id: ${response.id} and public key: ${response.generatedPublicKey}`);
+
+    const {
+      ciphertext: solanaCipherText,
+      dataToEncryptHash: solanaDataToEncryptHash,
+    } = await getEncryptedKey({
+      pkpSessionSigs,
+      litNodeClient,
+      id: response.id,
+    });
+
+    const accessControlConditions: AccessControlConditions = [
+      {
+        contractAddress: "",
+        standardContractType: "",
+        chain: "ethereum",
+        method: "",
+        parameters: [":userAddress"],
+        returnValueTest: {
+          comparator: "=",
+          value: pkpInfo.ethAddress!,
+        },
+      },
+    ];
```

Comment:
Probably makes more sense to say the prompt call only be decrypted within a specific LA